invenio_users_resources/records/api.py

```diff
@@ -255,55 +305,131 @@ def avatar_color(self):
         )
         return colors[int(normalized_group_initial, base=36) % len(colors)]
 
-    @property
-    def role(self):
-        """Cache for the associated role object."""
-        role = self._role
-        if role is None:
-            if role is None and self.id is not None:
-                role = current_datastore.find_role(self.id)
+    @classmethod
+    def get_record(cls, id_):
+        """Get the user group via the specified ID."""
+        # TODO how do we want to resolve the roles? via ID or name?
+        with db.session.no_autoflush:
+            role = current_datastore.role_model.query.get(id_)
+            if role is None:
+                return None
+            return cls.from_model(role)
+
+    @classmethod
+    def get_record_by_name(cls, name):
+        """Get the user group via the specified ID."""
+        # TODO how do we want to resolve the roles? via ID or name?
+        with db.session.no_autoflush:
+            role = current_datastore.role_model.query.filter_by(name=name).one_or_none()
+            if role is None:
+                return None
+            return cls.from_model(role)
 
-            self._role = role
 
-        return role
+class OrgNameDumperExt(SearchDumperExt):
+    """Custom fields dumper extension."""
 
-    def commit(self):
-        """Update the aggregate data on commit."""
-        # TODO this does not allow us to set properties via the aggregate?
-        #      because everything's taken from the Role object...
-        data = parse_role_data(self.role)
-        self.update(data)
-        self.model.update(data)
-        return self
+    def dump(self, record, data):
+        """Dump for faceting."""
+        org = data.get("org", None)
+        if org is not None and len(org) > 0:
```

Comment:
```suggestion
        if org and len(org) > 0:
```
Could `if org` suffice, as None and empty collections/lists are falsy?